### ISSUE TITLE
Disagreement with other ROC packages

### ISSUE THREAD

**Issue by IanButterworth** [diegozea/ROC.jl#7]
I thought it would be good to separate this issue out from the v1 changes.

To hightlight the issue on a simple dataset, where 5&6 are flipped, this is what the #6 (v1_changes_only) PR gives:
```julia
using ROC

scores = [1,2,3,4,6,5,7,8,9,10]
labels = [0,0,0,0,0,1,1,1,1,1]

res = roc(scores, labels,1)

using PyPlot
axhline(0,color="k",linewidth=0.1);axhline(1,color="k",linewidth=0.1)
axvline(0,color="k",linewidth=0.1);axvline(1,color="k",linewidth=0.1)
plot(res.FPR,res.TPR,"o-")
xlim(-0.1,1.1);xlim(-0.1,1.1)
title(string("AUC: ",AUC(res)))
```
![image](https://user-images.githubusercontent.com/1694067/56051389-68c94880-5d1c-11e9-88b6-cc0dde0fbd8c.png)

For **the same dataset**, [this online tool](http://www.rad.jhmi.edu/jeng/javarad/roc/JROCFITi.html) gives AUC of 0.96 (which I believe is correct):

![image](https://user-images.githubusercontent.com/1694067/56051428-91e9d900-5d1c-11e9-8b66-806465b3a9c8.png)

**Comment by diegozea** [comment]
Yes, extremes are missing now. I think that we can create a field `thresholds`/`cutoffs` with the unique values of the sorted `scores` + `typemax`. In that way, we are going to generate all the points by iterating the threshold vector and using `<` for the comparison instead of the actual subsetting strategy.

**Comment by IanButterworth** [comment]
Sounds good. Feel free to disregard my PR as there are many unnecessary changes in there now

**Comment by diegozea** [comment]
I think I will merge it and continue from there.

**Comment by IanButterworth** [comment]
Ok but wait, I’m working on a bug I found

**Comment by IanButterworth** [comment]
Ok, I was misreading the AUC on the external example. It should be 0.96 (now corrected above).

#4 now gives:
![image](https://user-images.githubusercontent.com/1694067/56056707-f2801280-5d2a-11e9-9f5f-b558f3b480a5.png)

@diegozea I *believe* you can merge, but you might want to check with some of your own test data first?

**Comment by diegozea** [comment]
I've changed the code to store and use a `thresholds` vector. It works fine with this example, but more testing is needed.
